src/sedcli_main.c

```diff
@@ -329,10 +329,30 @@ static int get_password(struct sed_key *pwd)
 	return 0;
 }
 
+static void print_sed_status(int status)
+{
+	const char *sed_status = NULL;
+
+	sed_status = sed_error_text(status);
+
+	if (sed_status == NULL && status != 0) {
+		if (status == -EINVAL) {
+			sedcli_printf(LOG_ERR, "Invalid parameter\n");
+		} else if (status == -ENODEV) {
+			sedcli_printf(LOG_ERR, "Couldn't determine device state\n");
+		} else if (status == -ENOMEM) {
+			sedcli_printf(LOG_ERR, "No memory\n");
+		} else {
+			sedcli_printf(LOG_ERR, "Unknown error\n");
+		}
+	} else {
+		sedcli_printf(LOG_INFO, "%s\n", sed_status);
```

Comment:
Last minute realization is that only successful status should be printed to stdout, other statuses (i.e errors) should be printed to stderr

Reply:
I think other messages like 'Invalid Parameter', 'Not Authorised' or 'SP Busy' etc need not have to be errors, they are just indicating the Driver State, its saying that the SP is busy, or the user is not authorized. The code isn't returning any error right..?

Reply:
> I think other messages like 'Invalid Parameter', 'Not Authorised' or 'SP Busy' etc need not have to be errors, they are just indicating the Driver State, its saying that the SP is busy, or the user is not authorized. The code isn't returning any error right..?

TCG_Storage_Architecture_Core_Spec_v2.01_r1.00; chapter 5.1.5 defines status code and its interpretation. I read Not authorized as an error/

Reply:
done